src/Components/Analyzers/src/ComponentParameterAnalyzer.cs

```diff
@@ -61,13 +62,20 @@ public override void Initialize(AnalysisContext context)
                     {
                         var captureUnmatchedValuesParameters = new List<IPropertySymbol>();
 
-                            // Per-property validations
-                            foreach (var property in properties)
+                        // Per-property validations
+                        foreach (var property in properties)
                         {
-                            if (property.SetMethod?.DeclaredAccessibility == Accessibility.Public)
+                            if (property.DeclaredAccessibility != Accessibility.Public)
                             {
                                 context.ReportDiagnostic(Diagnostic.Create(
-                                    DiagnosticDescriptors.ComponentParametersShouldNotBePublic,
+                                    DiagnosticDescriptors.ComponentParametersShouldBePublic,
+                                    property.Locations[0],
+                                    property.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat)));
+                            }
+                            else if (property.SetMethod?.DeclaredAccessibility != Accessibility.Public)
```

Comment:
`else if` so we don't flood the user with errors when they have:
```C#
[Parameter] int Foo { get; set; }
```
Because the setter is technically private.